docs/hooks.md

```diff
@@ -0,0 +1,172 @@
+---
+layout: default
+title: Hooks
+nav_order: 7
+has_children: false
+---
+
+# Configurable Hooks
+Like other version control systems, lakeFS allows the configuration of `Actions` to trigger when predefined events occur.
+ 
+Supported Events:
+1. `pre_commit` - Action runs when the commit occurs, before the commit is finalized.
+2. `pre_merge` - Action runs when the merge occurs, before the merge is finalized.
+ 
+lakeFS `Actions` are handled per repository and cannot be shared between repositories.  
+Failure of any `Hook` under any `Action` of a `pre_*` event will result in aborting the lakeFS operation that is taking place.
+
+`Hooks` are managed by `Action` files that are written to a prefix in the lakeFS repository. 
+This allows configuration-as-code inside lakeFS, where `Action` files are declarative and written in YAML.
+
+## Example use-cases
+1. Format Validator:
+   A webhook that checks new files to ensure they are of a set of allowed data format.
+2. Schema Validator:
+   A webhook that reads new Parquet and ORC files to ensure they don't contain a block list of column names (or name prefixes). 
+   This is useful when we want to avoid accidental PII exposure.
+
+For more examples and configuration samples, check out [lakeFS-hooks](https://github.com/treeverse/lakeFS-hooks) example repo.
+
+## Action
+An `Action` is a list of `Hooks` with the same trigger configuration, i.e. an event will trigger all `Hooks` under an `Action`, or none at all.
+The `Hooks` under an `Action` are ordered, and a `Hook` will only be executed if all previous `Hooks` had passed.
+
```

Comment:
These two sentences are contradictory

Reply:
I understand why it's confusing: trigger != execution .
Added clarification that execution depends on failures, not triggering.